ocaml/tests/test_repository_helpers.ml

```diff
@@ -511,551 +511,1001 @@ module EvalGuidanceForOneUpdate = Generic.MakeStateless (struct
         str "%a"
           Dump.(
             pair
-              (list (pair string (record @@ fields_of_updateinfo)))
-              (record @@ fields_of_update)
+              (pair
+                 (list (pair string (record @@ fields_of_updateinfo)))
+                 (record @@ fields_of_update)
+              )
+              (list string)
           )
       )
 
     let string_of_output_t g =
       Fmt.(str "%a" Dump.(string)) (UpdateInfo.guidance_to_string g)
   end
 
-  let transform (updates_info, update) =
-    eval_guidance_for_one_update ~updates_info ~update ~kind:Guidance.Absolute
-      ~upd_ids_of_livepatches:UpdateIdSet.empty
+  let transform ((updates_info, update), upd_ids_of_livepatches) =
+    eval_guidance_for_one_update ~updates_info ~update
+      ~kind:Guidance.Recommended
+      ~upd_ids_of_livepatches:(UpdateIdSet.of_list upd_ids_of_livepatches)
```

Comment:
Could the changes here be explained in the commit message? These seem important and introduce quite a lot of churn in the test data, it would be good to know why

Reply:
Sure. I will add a commit message. The message would be:
Prior to support live patch, the guidance of an update is determined by "recommendedGuidance". With supporting to live patch, a new metadata "livepatchGuidance" is introduced. In an update, if a live patch in it is determined as applicable, its "livepatchGuidance" will overwrite its "recommendedGuidance".

Reply:
Looks like this hasn't yet been added to the commit message.